runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/StreamingEngineClientTest.java

```diff
@@ -196,8 +196,7 @@ public void testStreamsStartCorrectly() throws InterruptedException {
     String workerToken = "workerToken1";
     String workerToken2 = "workerToken2";
 
-    Thread streamingEngineClientThread =
-        new Thread(streamingEngineClient::waitForFirstWorkerMetadata);
+    Thread streamingEngineClientThread = new Thread(this::waitForFirstWorkerMetadata);
```

Comment:
can this thread be removed? You aren't joining it and now the wait itself doesn't trigger anything, so it seems like it is unnecessary.

ditto for the others, perhaps the waitForFirstWorkerMetadata can just be removed by relying on getWorkerMetadtadaReady instead